app/events/accounts/account_deleted_event.rb

```diff
@@ -5,7 +5,7 @@ def self.create(account)
       account_name: account.name,
       provider:     account.provider_account,
       buyer:        account.buyer,
-      user_id:      account.first_admin.try(:id),
+      user_id:      account.first_admin.try(:id) || account.first_admin_id,
```

Comment:
I would do in Account `alias_attribute :first_admin_id_on_provider_signup, :first_admin_id`

And somehow add a comment to this PR

Reply:
Well, you already added the comment to the PR 😄 https://github.com/3scale/porta/pull/8#issuecomment-446162555